vars/retrievePreviousBuild.groovy

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+/** Library to download previous build artifacts used for incremental build.
+ @param Map args = [:] args A map of the following parameters
+ @param args.inputManifest <required> - Relative path to input manifest containing all the components to build.
+ @param args.jobName <optional> - Relative path to input manifest containing all the components to build.
+ @param args.platform <required> - Platform of previous build to retrieve.
+ @param args.architecture <required> - Architecture of previous build to retrieve.
+ @param args.distribution <required> - Distribution of previous build to retrieve.
+ @param args.previousBuildId <optional> - Build id of previous build for incremental build. Defaults to latest.
```

Comment:
I think you forgot to assign the default value. if the value is null or empty please assign `previousBuildId = latest`

Reply:
This default value will not be assigned here right? I think the default should be assigned at Job level when we initiate the parameter.

Reply:
Had an offline discussion with @gaiksaya. Just updated a new line for defaulting this `previousBuildId` to `latest` if user doesn't provide argument.